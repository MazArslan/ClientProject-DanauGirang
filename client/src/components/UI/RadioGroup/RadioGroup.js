import React from "react";
import propTypes from "prop-types";
import { Row, Col, FormGroup, ButtonGroup, Button } from "reactstrap";

/**
 * Radio group component
 * Props:
 *    aiId: unique identifier for the group
 *    radioButtonForm: an object used to store the state generated by the buttons
 *    selectedButtonChanged: a function that handles updating the radioButtonForm state
 *    when a different button is selected by the user
 *  @param { aiId, radioButtonForm, selectedButtonChanged } props
 */
const radioGroup = props => {
  const { aiId, radioButtonForm, selectedButtonChanged } = props;

  return (
    <FormGroup tag="fieldset" row>
      <Row>
        <Col className="text-center">
          <ButtonGroup>
            <Button
              color="primary"
              outline
              value="YES"
              onClick={selectedButtonChanged}
              active={radioButtonForm[aiId] === "YES"}
            >
              Yes
            </Button>
            <Button
              color="primary"
              outline
              value="NO"
              onClick={selectedButtonChanged}
              active={radioButtonForm[aiId] === "NO"}
            >
              No
            </Button>
          </ButtonGroup>
        </Col>
      </Row>
    </FormGroup>
  );
};

radioGroup.propTypes = {
  aiId: propTypes.string.isRequired,
  radioButtonForm: propTypes.object.isRequired,
  selectedButtonChanged: propTypes.func.isRequired
};

export default radioGroup;
