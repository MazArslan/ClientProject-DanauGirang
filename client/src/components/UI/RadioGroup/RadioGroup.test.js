import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import RadioGroup from "./RadioGroup";

Enzyme.configure({ adapter: new Adapter() });

describe("<RadioGroup/>", () => {
  const onActionMock = jest.fn();
  const props = {
    aiId: "5e8f013965372f3768b7a2c3",
    radioButtonForm: { "5e8f013965372f3768b7a2c3": "YES" },
    selectedButtonChanged: onActionMock
  };
  const container = shallow(<RadioGroup {...props} />);

  describe("component rendering", () => {
    it("should render correctly", () => {
      expect(container.exists()).toBe(true);
      expect(toJson(container)).toMatchSnapshot();
    });

    it("should render active YES button", () => {
      expect(
        container
          .find("Button")
          .at(0)
          .props().value
      ).toEqual("YES");
      expect(
        container
          .find("Button")
          .at(0)
          .props().active
      ).toEqual(true);
    });

    it("should render inactive NO button", () => {
      expect(
        container
          .find("Button")
          .at(1)
          .props().value
      ).toEqual("NO");
      expect(
        container
          .find("Button")
          .at(1)
          .props().active
      ).toEqual(false);
    });
  });
});
