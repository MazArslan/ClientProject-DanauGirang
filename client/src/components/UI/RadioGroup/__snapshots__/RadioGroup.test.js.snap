// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup/> component rendering should render correctly 1`] = `
<FormGroup
  row={true}
  tag="fieldset"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="text-center"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <ButtonGroup
        role="group"
        tag="div"
      >
        <Button
          active={true}
          color="primary"
          onClick={[MockFunction]}
          outline={true}
          tag="button"
          value="YES"
        >
          Yes
        </Button>
        <Button
          active={false}
          color="primary"
          onClick={[MockFunction]}
          outline={true}
          tag="button"
          value="NO"
        >
          No
        </Button>
      </ButtonGroup>
    </Col>
  </Row>
</FormGroup>
`;
